### PR TITLE
Merge exported contents into existing files

### PR DESCRIPTION
**Pull request by joao-borges**
- updating existing entities
- adding new ones

Fixes [46](https://github.com/ca-api-gateway/gateway-developer-plugin/issues/46)

### CODE REVIEW

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 1]
```diff
@@ -8,16 +8,23 @@
 
 import com.fasterxml.jackson.core.JsonFactory;
 import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.core.JsonParser'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgO&open=AWYScDjcM63is4RDDPgO)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgO

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/loader/ListenPortLoader.java [hunk 2]
```diff
@@ -13,6 +13,7 @@
 import javax.inject.Singleton;
 import java.util.List;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/loader/ListenPortLoader.java]: Code Smell Code Smell: Remove this unused import 'java.util.List'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDeJM63is4RDDPgL&open=AWYScDeJM63is4RDDPgL)

[ISSUE_KEY]: AWYScDeJM63is4RDDPgL

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 3]
```diff
@@ -8,16 +8,23 @@
...
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
 
 import java.io.IOException;
 import java.util.HashMap;
+import java.util.List;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'java.util.List'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgS&open=AWYScDjcM63is4RDDPgS)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgS

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 4]
```diff
@@ -8,16 +8,23 @@
 
 import com.fasterxml.jackson.core.JsonFactory;
 import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import com.fasterxml.jackson.databind.module.SimpleModule;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.databind.module.SimpleModule'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgR&open=AWYScDjcM63is4RDDPgR)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgR

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 5]
```diff
@@ -8,16 +8,23 @@
 
 import com.fasterxml.jackson.core.JsonFactory;
 import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.core.JsonProcessingException'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgP&open=AWYScDjcM63is4RDDPgP)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgP

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 6]
```diff
@@ -8,16 +8,23 @@
...
 import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.databind.deser.std.StdDeserializer'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgQ&open=AWYScDjcM63is4RDDPgQ)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgQ

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/beans/ListenPort.java [hunk 7]
```diff
@@ -8,20 +8,21 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 
-import java.util.List;
+import java.util.Set;
 import java.util.Map;
+import java.util.Set;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/beans/ListenPort.java]: Code Smell Code Smell: Remove this duplicated import. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDilM63is4RDDPgM&open=AWYScDilM63is4RDDPgM)

[ISSUE_KEY]: AWYScDilM63is4RDDPgM

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 8]
```diff
@@ -8,16 +8,23 @@
...
+import com.fasterxml.jackson.databind.module.SimpleModule;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
 import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.dataformat.yaml.YAMLGenerator'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgN&open=AWYScDjcM63is4RDDPgN)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgN

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/beans/ListenPort.java [hunk 9]
```diff
@@ -8,8 +8,8 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 
-import java.util.List;
 import java.util.Map;
+import java.util.Set;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/beans/ListenPort.java]: Code Smell Code Smell: Remove this duplicated import. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDilM63is4RDDPgM&open=AWYScDilM63is4RDDPgM)

[ISSUE_KEY]: AWYScDilM63is4RDDPgM

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 10]
```diff
@@ -10,14 +10,15 @@
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.*;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
-import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
 
 import java.io.IOException;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.databind.deser.std.StdDeserializer'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgQ&open=AWYScDjcM63is4RDDPgQ)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgQ

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 11]
```diff
@@ -10,14 +10,15 @@
...
 import java.util.HashMap;
 import java.util.Map;
 import java.util.logging.Level;
 import java.util.logging.Logger;
 
+import static com.fasterxml.jackson.dataformat.yaml.YAMLGenerator.Feature.WRITE_DOC_START_MARKER;
+
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'java.util.List'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgS&open=AWYScDjcM63is4RDDPgS)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgS

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 12]
```diff
@@ -10,14 +10,15 @@
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.*;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.core.JsonProcessingException'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgP&open=AWYScDjcM63is4RDDPgP)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgP

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/loader/ListenPortLoader.java [hunk 13]
```diff
@@ -11,8 +11,8 @@
 import org.w3c.dom.Element;
 
 import javax.inject.Singleton;
-import java.util.List;
 import java.util.Map;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/loader/ListenPortLoader.java]: Code Smell Code Smell: Remove this unused import 'java.util.List'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDeJM63is4RDDPgL&open=AWYScDeJM63is4RDDPgL)

[ISSUE_KEY]: AWYScDeJM63is4RDDPgL

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 14]
```diff
@@ -10,14 +10,15 @@
...
-import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
 
 import java.io.IOException;
 import java.util.HashMap;
 import java.util.Map;
 import java.util.logging.Level;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.dataformat.yaml.YAMLGenerator'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgN&open=AWYScDjcM63is4RDDPgN)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgN

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 15]
```diff
@@ -10,14 +10,15 @@
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.*;
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
-import com.fasterxml.jackson.dataformat.yaml.YAMLGenerator;
 
 import java.io.IOException;
 import java.util.HashMap;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.databind.module.SimpleModule'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgR&open=AWYScDjcM63is4RDDPgR)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgR

File: src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java [hunk 16]
```diff
@@ -10,14 +10,15 @@
 import com.fasterxml.jackson.core.JsonGenerator;
 import com.fasterxml.jackson.databind.*;
```
Comment (apigatewayopensource) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/util/json/JsonTools.java]: Code Smell Code Smell: Remove this unused import 'com.fasterxml.jackson.core.JsonParser'. ([squid:UselessImportCheck](https://sonarcloud.io/organizations/ca-api-gateway/rules#rule_key=squid%3AUselessImportCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=com.ca.apim.gateway%3Agateway-export-plugin&pullRequest=27&issues=AWYScDjcM63is4RDDPgO&open=AWYScDjcM63is4RDDPgO)

[ISSUE_KEY]: AWYScDjcM63is4RDDPgO

**Comment by joao-borges** [comment]
@kazvictor I have not found any library that loads properties ordered.
Most answers on google end on someone writing a subclass of `Properties` using an internal structure that handle the ordering, but leaving the load/store by current implementation from superclass.
That's what I did, but I am not a fan of this solution. I did not find a better solution.

File: src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/WriterHelper.java [hunk 17]
```diff
@@ -102,4 +153,177 @@ static void writePropertiesFile(File rootFolder, DocumentFileUtils documentFileU
         // keeping original order if any
         return new LinkedHashMap<>(originalMap);
     }
+
+    /**
+     * Internal class used to read the properties file. Takes advantage of the existing default mechanism of reading and parsing properties files,
+     * but stores them in a ordered structure. This is done in order to keep original order from the properties file.
+     */
+    private static class OrderedProperties extends Properties {
```
Comment (kazvictor) [inline src/main/java/com/ca/apim/gateway/cagatewayexport/tasks/explode/writer/WriterHelper.java]: Move this to its own file. The WriteHelper is getting too long.

**Pull request merged**